packages/protocol/schemaCodegen.js

```diff
@@ -16,29 +15,35 @@ if (args.length > 0) {
     }
     exit(0);
   }
-  schema = args[0];
+  schemaUri = args[0];
 }
 
-function fetchSchema() {
+function loadJson(uri) {
   const http = require('https');
-  const file = fs.createWriteStream(schemaFile);
-  http.get(schema, function (response) {
-    response.pipe(file);
-    file.on('finish', () => {
-      file.close();
-      console.log('Downloaded ' + schema);
-      codegen();
-    });
-  });
+  console.log(`loading ${uri}`);
+  const download = new Promise(result =>
+    http.get(uri, res => {
```

Comment:
maybe you can use also the "fetch" method here? but I'm not sure :)

Reply:
unfortunately not, with node 17 as preview, and starting from node 18 it would work. but we are on 16.20

Reply:
Ahh ok :see_no_evil:, wasn't sure if its already supported in node ;)